Sources/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -411,4 +411,373 @@ extension FileDescriptor {
+    public var teardownSequence: [TeardownStep] = []
+
+    public init() {}
+}
+
+extension PlatformOptions: CustomStringConvertible, CustomDebugStringConvertible {
+    internal func description(withIndent indent: Int) -> String {
+        let indent = String(repeating: " ", count: indent * 4)
+        return """
+            PlatformOptions(
+            \(indent)    userID: \(String(describing: userID)),
+            \(indent)    groupID: \(String(describing: groupID)),
+            \(indent)    supplementaryGroups: \(String(describing: supplementaryGroups)),
+            \(indent)    processGroupID: \(String(describing: processGroupID)),
+            \(indent)    createSession: \(createSession)
+            \(indent))
+            """
+    }
+
+    public var description: String {
+        return self.description(withIndent: 0)
+    }
+
+    public var debugDescription: String {
+        return self.description(withIndent: 0)
+    }
+}
+
+// Special keys used in Error's user dictionary
+extension String {
+    static let debugDescriptionErrorKey = "DebugDescription"
+}
+
+internal func pthread_create(_ body: @Sendable @escaping () -> ()) throws(SubprocessError.UnderlyingError) -> pthread_t {
+    final class Context {
+        let body: @Sendable () -> ()
+        init(body: @Sendable @escaping () -> Void) {
+            self.body = body
+        }
+    }
+    #if canImport(Glibc) || canImport(Musl)
+    func proc(_ context: UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer? {
+        (Unmanaged<AnyObject>.fromOpaque(context!).takeRetainedValue() as! Context).body()
+        return nil
+    }
+    #elseif canImport(Bionic)
+    func proc(_ context: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer {
+        (Unmanaged<AnyObject>.fromOpaque(context).takeRetainedValue() as! Context).body()
+        return context
+    }
+    #endif
+    #if (os(Linux) && canImport(Glibc)) || canImport(Bionic)
+    var thread = pthread_t()
+    #else
+    var thread: pthread_t?
+    #endif
+    let rc = pthread_create(
+        &thread,
+        nil,
+        proc,
+        Unmanaged.passRetained(Context(body: body)).toOpaque()
+    )
+    if rc != 0 {
+        throw SubprocessError.UnderlyingError(rawValue: rc)
+    }
+    #if (os(Linux) && canImport(Glibc)) || canImport(Bionic)
+    return thread
+    #else
+    return thread!
+    #endif
+}
+
+#endif  // !canImport(Darwin)
+
+extension ProcessIdentifier {
+    /// Reaps the zombie for the exited process. This function may block.
+    @available(*, noasync)
+    internal func blockingReap() throws(SubprocessError.UnderlyingError) -> TerminationStatus {
+        try _blockingReap(pid: value)
+    }
+
+    /// Reaps the zombie for the exited process, or returns `nil` if the process is still running. This function will not block.
+    internal func reap() throws(SubprocessError.UnderlyingError) -> TerminationStatus? {
+        try _reap(pid: value)
+    }
+}
+
+@available(*, noasync)
```

Comment:
Just curious, what's this for?

Reply:
Semantically labels it as blocking -- you'll get an error if you accidentally try to call it in an `async` context.